src/js/WinJS/Accents.ts

```diff
@@ -0,0 +1,171 @@
+// Copyright (c) Microsoft Corporation.  All Rights Reserved. Licensed under the MIT License. See License.txt in the project root for license information.
+import _Global = require("./Core/_Global");
+import _WinRT = require("./Core/_WinRT")
+
+import _Base = require("./Core/_Base");
+import _BaseUtils = require("./Core/_BaseUtils");
+
+var Constants = {
+    accentStyleId: "WinJSAccentsStyle",
+    themeDetectionTag: "winjs",
+
+    hoverSelector: "html.win-hoverable",
+    lightThemeSelector: ".win-ui-light",
+    darkThemeSelector: ".win-ui-dark"
+};
+
+var CSSSelectorTokens = [".", "#", ":"];
+
+var UISettings: _WinRT.Windows.UI.ViewManagement.UISettings = null;
+var colors: string[] = [];
+var isDarkTheme = false;
+var rules: { selector: string; props: { name: string; value: ColorTypes; }[]; noHoverSelector: boolean; }[] = [];
+var writeRulesTOHandle = -1;
+
+export enum ColorTypes {
+    accent = 0,
+    listSelectRest = 1,
+    listSelectHover = 2,
+    listSelectPress = 3,
+    listSelectRestInverse = 4,
+    listSelectHoverInverse = 5,
+    listSelectPressInverse = 6,
+}
+
+export function createAccentRule(selector: string, props: { name: string; value: ColorTypes; }[], noHoverSelector = false) {
+    rules.push({ selector: selector, props: props, noHoverSelector: noHoverSelector });
+    scheduleWriteRules();
+}
+
+function scheduleWriteRules() {
+    if (!rules.length || writeRulesTOHandle !== -1) {
+        return;
+    }
+    writeRulesTOHandle = _BaseUtils._setImmediate(() => {
+        writeRulesTOHandle = -1;
+        cleanup();
+
+        var inverseThemeSelector = isDarkTheme ? Constants.lightThemeSelector : Constants.darkThemeSelector;
+
+        var style = _Global.document.createElement("style");
+        style.id = Constants.accentStyleId;
+        style.textContent = rules.map(rule => {
+            // example rule: { selector: "  .foo,  .bar:hover ,  div:hover  ", props: [{ name: "color", value: 0 }, { name: "background-color", value: 1 }, noHoverSelector: false }
+
+            var body = "  " + rule.props.map(prop => prop.name + ": " + colors[prop.value] + ";").join("\n  ");
+            // body = color: *accent*; background-color: *listSelectHover*
+
+            var selectorSplit = rule.selector.split(",").map(str => str.trim()); // [".foo", ".bar:hover", "div"]
+            var selector = selectorSplit.join(",\n"); // ".foo, .bar:hover, div"
+
+            // Hover Selectors
+            if (!rule.noHoverSelector) {
```

Comment:
Remove. Have developer specify `html.win-hoverable` manually.

Make sure to handle inverse theme correctly with `html.win-hoverable` (i.e. `win-ui-light` class has to come after the `html.win-overable` and not before)